overlord/ifacestate/helpers.go

```diff
@@ -159,7 +159,10 @@ func (m *InterfaceManager) regenerateAllSecurityProfiles() error {
 		}
 	}
 
-	return interfaces.WriteSystemKey()
+	if err := interfaces.WriteSystemKey(); err != nil {
```

Comment:
This is fine, I checked that SystemKeyMismatch() already handles errors gracefully.